pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -350,6 +352,7 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 
 	for _, rprt := range rprts {
 		if rprt != nil {
+			go c.timeoutHandler.WaitPipelineRun(pr)
```

Comment:
i'm a bit confused, why do we need to call `go WaitPipelineRun` for every new taskrun created for the PipelineRun? I would assume we only needed to call `go WaitPipelineRun` once when the PipelineRun started. (I'm probably totally wrong, if so then maybe a comment here would help me understand!)

Reply:
>  I would assume we only needed to call go WaitPipelineRun once when the PipelineRun started.

Yes. There is no specific place in reconciler code where it checks if its new or existing pipelinerun.

Reply:
hm doesnt this mean we'll end up with more goroutines tracking pipelineruns than we need?

Reply:
oh indeed, it should be just like task run below

```go
go c.timeoutHandler.WaitPipelineRun(pr)
for _, rprt := rang rprts {
// […]
```

Reply:
I updated this logic to use "hasstarted" function. WDYT @vdemeester ?

Reply:
thanks @shashwathi !! ❤️